File: src/main/java/com/example/global/config/AsyncConfig.java
package com.example.global.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;

import java.util.concurrent.Executor;

@Configuration
@EnableAsync
public class AsyncConfig {
    @Bean(name = "festivalDetailTaskExecutor")
    public Executor festivalDetailTaskExecutor() {
        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();

        // 기본 실행 스레드 수
        executor.setCorePoolSize(2);

        // 최대 실행 스레드 수
        executor.setMaxPoolSize(4);

        // 대기 큐 크기
        executor.setQueueCapacity(100);

        // 스레드 이름 prefix
        executor.setThreadNamePrefix("festival-detail-");

        executor.initialize();
        return executor;
    }
}

File: src/main/kotlin/com/example/global/config/AsyncConfig.kt
package com.example.global.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableAsync
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor
import java.util.concurrent.Executor

@Configuration
@EnableAsync
class AsyncConfig {

    @Bean(name = ["festivalDetailTaskExecutor"])
    fun festivalDetailTaskExecutor(): Executor =
        ThreadPoolTaskExecutor().apply {
            corePoolSize = 2 // 기본 실행 스레드 수
            maxPoolSize = 4 // 최대 실행 스레드 수
            queueCapacity = 100 // 대기 큐 크기
            setThreadNamePrefix("festival-detail-") // 스레드 이름 prefix
            initialize()
        }
}